private/buf/cmd/buf/buf_test.go

```diff
@@ -294,10 +294,13 @@ func TestFail7(t *testing.T) {
 		t,
 		nil,
 		bufctl.ExitCodeFileAnnotation,
-		"", // stdout is empty
+		filepath.FromSlash(`testdata/fail/buf/buf.proto:3:1:Files with package "other" must be within a directory "other" relative to root but were in directory "fail/buf".
+testdata/fail/buf/buf.proto:6:9:Field name "oneTwo" should be lower_snake_case, such as "one_two".`),
 		// This is new behavior we introduced. When setting a config override, we no longer do
 		// a search for the controlling workspace. See bufctl/option.go for additional details.
-		filepath.FromSlash(`Failure: testdata/export/other/proto/unimported.proto: import "another.proto": file does not exist`),
+		// Only the paths specified in the command are considered.
+		// This avoids build failures from other proto files under testdata.
```

Comment:
This test showcases the failure which is no longer hit with the new pathing restriction. The file `testdata/export/other/proto/unimported.proto` imports are no longer resolved.

Reply:
This feels strictly wrong - if `buf build` would not pass, `buf lint` should not pass either. We probably should talk about this live.

Reply:
`buf build` does pass for this case, `buf lint` did not pass before Ed's change because of the full imports check. We can chat about it live, for sure. It might be worth clarifying in the test comment that `buf build --path testdata/fail/buf/buf.proto` does pass.

Reply:
Yep will clarify, build failure is on linting not running build. `buf build` passes.